papermario/src/code_fa4c0_len_3bf0.c

```diff
@@ -2,35 +2,130 @@
+}
+
+s32 SetMusicTrack(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 musicPlayer = get_variable(script, *ptrReadPos++);
+    s32 songID = get_variable(script, *ptrReadPos++);
+    s32 variation = get_variable(script, *ptrReadPos++);
+    s32 volume = get_variable(script, *ptrReadPos++);
+
+    return (set_music_track(musicPlayer, songID, variation, 0x1F4, (volume << 0x10) >> 0x10) != 0) * 2;
+}
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, FadeInMusic);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, PlayAmbientSounds);
+s32 PlayAmbientSounds(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+
+    return (play_ambient_sounds(soundID, 0xFA) != 0) * 2;
+}
+
+s32 PlaySound(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+
+    play_sound(soundID);
+    return 2;
+}
+
+s32 func_802D617C(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+    s32 value2 = get_variable(script, *ptrReadPos++);
+
+    _play_sound(soundID, value2 & 0xFF, 0, 0);
+    return 2;
+}
+
+s32 PlaySoundAt(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+    s32 value2 = get_variable(script, *ptrReadPos++);
+    s32 x = get_variable(script, *ptrReadPos++);
+    s32 y = get_variable(script, *ptrReadPos++);
+    s32 z = get_variable(script, *ptrReadPos++);
+
+    play_sound_at_position(soundID, value2, (f32) x, (f32) y, (f32) z);
+    return 2;
+}
+
+s32 StopSound(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+
+    stop_sound(get_variable(script, *ptrReadPos++));
+    return 2;
+}
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, PlaySound);
+s32 func_802D62E4(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, PlaySoundAt);
+    func_80149A6C(get_variable(script, *ptrReadPos++), 1);
+    return 2;
+}
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, StopSound);
+s32 UseDoorSounds(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, UseDoorSounds);
+    gCurrentDoorSoundsSet = get_variable(script, *ptrReadPos++);
+    return 2;
+}
+
+s32 func_802D6340(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+
+    D_80151308 = get_variable(script, *ptrReadPos++);
+    return 2;
+}
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, PlaySoundAtF);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, RemoveKeyItemAt);
+s32 RemoveKeyItemAt(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 index = get_variable(script, *ptrReadPos++);
+    s16* ptrTemp = D_8010F304;
+    ptrTemp[index] = 0;
```

Comment:
Please separate variable declarations from the rest of the function with an empty line